transformers4rec/torch/utils/data_utils.py

```diff
@@ -400,17 +408,25 @@ def _augment_schema(
             schema[label] = schema[label].with_tags(Tags.CATEGORICAL)
         for label in conts:
             schema[label] = schema[label].with_tags(Tags.CONTINUOUS)
+        for col in lists:
+            schema[col] = schema[col].with_tags(Tags.LIST)
 
         return schema
 
-    def set_dataset(self, buffer_size, engine, reader_kwargs):
+    def set_dataset(self, buffer_size, engine, reader_kwargs, schema=None):
         dataset = validate_dataset(
             self.paths_or_dataset,
             self.batch_size,
             buffer_size,
             engine,
             reader_kwargs,
         )
+        if schema:
+            # set the dataset schema based on the provided one to keep all original information
+            if isinstance(schema, Schema):
+                # convert merlin-standardlib schema to merlin-core schema
+                schema = to_core_schema(schema)
+            dataset.schema = schema
```

Comment:
This is needed to ensure the original properties of the schema the user provides are kept. It is required in the PR as we use the data loader's output schema to define the model.